File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.java
import java.util.Scanner;

public class Main {
    static int n, m;
    static int[] arr;

    public static void goMan(int b) {
        for (int i = 1; i * b <= n; i++) {
            arr[i * b] ^= 1;
        }
    }

    public static void goWo(int b) {
        arr[b] ^= 1;
        for (int i = 1; (b - i >= 1 && b + i <= n); i++) {
            if (arr[b - i] != arr[b + i]) return;
            arr[b - i] ^= 1;
            arr[b + i] ^= 1;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        n = scanner.nextInt();
        arr = new int[n + 1];

        for (int i = 1; i <= n; i++) {
            arr[i] = scanner.nextInt();
        }

        m = scanner.nextInt();
        for (int i = 0; i < m; i++) {
            int a = scanner.nextInt();
            int b = scanner.nextInt();
            if (a == 1) {
                goMan(b);
            } else {
                goWo(b);
            }
        }

        for (int i = 1; i <= n; i++) {
            System.out.print(arr[i] + " ");
            if (i % 20 == 0) System.out.println();
        }
    }
}

File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.cc
#include <bits/stdc++.h>
using namespace std;
int n, arr[104],m,a,b;


void goMan(int b){
    for(int i=1; i*b <= n; i++){
        arr[i*b] ^= 1;
    }
}

void goWo(int b){
    arr[b] ^= 1;
    for(int i=1; (b-i >= 1 && b+i <= n) ; i++ ){
        if(arr[b-i] != arr[b+i]) return;
        arr[b-i]  ^= 1;
        arr[b+i] ^= 1;
    }
}
int main() {

    cin >> n;
    for(int i=1;i<=n;i++) {
        cin >> arr[i];
    }
    cin >> m;
    for(int i=0;i<m;i++){
        cin >> a >> b;
        if(a == 1) {
            goMan(b);
        }
        else {
            goWo(b);
        }
    }

    for(int i=1;i<=n;i++) {
        cout << arr[i] <<" ";
        if(i%(20) ==0) cout <<"\n";
    }
    return 0;
}

